ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java

```diff
@@ -197,4 +197,22 @@ default boolean addDynamicSplitPruningEdge(ExprNodeDesc syntheticFilterPredicate
   default Map<String, String> getOperatorDescProperties(OperatorDesc operatorDesc, Map<String, String> initialProps) {
     return initialProps;
   }
+
+  /**
+   * Return some basic statistics (numRows, numFiles, totalSize) calculated by the underlying storage handler
+   * implementation.
+   * @param tableDesc a valid table description, used to load the table
+   * @return map of basic statistics, can be null
+   */
+  default Map<String, String> getBasicStatistics(TableDesc tableDesc) {
+    return null;
+  }
+
+  /**
+   * Check if the storage handler can provide basic statistics.
+   * @return true if the storage handler can supply the basic statistics
+   */
+  default boolean canProvideBasicStatistics() {
```

Comment:
Do we need both methods?
Wouldn't it be better to handle `null` from `getBasicStatistics()` as `!canProvideBasicStatistics()`?

Reply:
Ok.. I see why it is separated...